### PR TITLE
PCDLoader decoder compatibility

### PR DESCRIPTION
**Pull request by yomotsu**
PCDLoader example doesn't work in IE and Edge. This P-R is to solve the problem.

PCD file can contain binary.

> DATA - specifies the data type that the point cloud data is stored in. As of version 0.7, two data types are supported: ascii and binary. See the next section for more details.

via http://pointclouds.org/documentation/tutorials/pcd_file_format.php

If the PCD file is binary version,  it causes a garbled and `decodeURIComponent` that is a fallback for [`TextDecorder` unsupported browsers](https://github.com/mrdoob/three.js/blob/dev/src/loaders/LoaderUtils.js#L28), such as IE and Edge will return "malformed error"

Fortunately, the header section is always ascii format, and `decodeURIComponent` and `escape` are unnecessary.

This patch is adding a decoder in PCDLoader, rather than using `THREE.LoaderUtils.decodeText`

### CODE REVIEW

**Comment by donmccurdy** [comment]
It would be nice to patch `LoaderUtils.decodeText` for this case somehow. Perhaps:

```js
LoaderUtils.decodeText( array, isASCII = true );
```

Using `decodeURIComponent( escape( s ) );` for multi-byte characters was always a bit of a hack for Edge+IE11. Edge is [working toward TextDecoder](https://developer.microsoft.com/en-us/microsoft-edge/platform/status/encodingstandard/) support now, or another option is to use the [text-encoding polyfill](https://github.com/inexorabletash/text-encoding).

**Comment by yomotsu** [comment]
@donmccurdy thanks for taking a look

Sounds the first option

> ```
> isASCII = false
> ```

is better at this moment?

I have also tried with the polyfill and it works but [the lib](https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js) is kinda big.

**Comment by yomotsu** [comment]
@donmccurdy 
I added `isASCII` option. Does it look good?

**Review by donmccurdy — APPROVED** [review]

**Comment by donmccurdy** [comment]
Looks good to me, thanks!

File: src/loaders/LoaderUtils.js [hunk 1]
```diff
@@ -24,6 +24,8 @@ var LoaderUtils = {
 
 		}
 
+		if ( isASCII ) return s;
+
 		// Merges multi-byte utf-8 characters.
 		return decodeURIComponent( escape( s ) );
```
Comment (Mugen87) [inline src/loaders/LoaderUtils.js]: Maybe:
```js
return ( isASCII === true ) ? s : decodeURIComponent( escape( s ) );
```

**Review by Mugen87 — CHANGES_REQUESTED** [review]

**Comment by Mugen87** [comment]
Please update the docs, too 😇 https://threejs.org/docs/index.html#api/loaders/LoaderUtils

**Comment by yomotsu** [comment]
@Mugen87 updated. but I'm not confident for the doc, since I'm not a english native speaker. I was wondering if someone could rewrite later if needed 🙇

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
Closing in favor of #16362

**Pull request closed without merging**